File: credit/credit.py
import re
N = int(input("Number:  "))

def credit(N):
  seconds = []
  rests = []
  arr = [int(x) for x in str(N)]
  i = len(arr) - 2
  while(i>=0):
    i -= 2
    arr[i] = arr[i]*2
    if arr[i]>10:
      seconds.append(int(str(arr[i])[0]))
      seconds.append(int(str(arr[i])[1]))
    else:
      seconds.append(arr[i])
  j = len(arr)-1
  while(j>=0):
    j -=2
    rests.append(arr[j])
  if (sum(seconds)+ sum(rests))%10 == 0:
    if re.match(r"^4", str(N)):
        print("VISA")
    elif re.match(r"^3[4|7]", str(N)):
        print("MASTERCARD")
    else:
        print("AMAX")
  else:
    print(False)


credit(N)

File: credit/credit.js
function credit(N) {
  let arr = Array.from(String(N));
  let seconds = [], rests = [];
  for(let i = arr.length-2; i>=0; i= i-2) {
		let second = arr[i];
    second = second * 2;
    (second > 10)?
      seconds.push(Number(String(second)[0]), 		Number(String(second)[1])):seconds.push(second);
  }
  let secondSum = seconds.reduce((a,b) => a+b);
  for(let j = arr.length-1; j>=0; j = j-2) {
    let rest = arr[j];
    rests.push(Number(rest));
  }
  let restSum = rests.reduce((a, b)=> a+b);
  if((restSum + secondSum)%10 === 0) {
    if(/^4/.test(N))
      return "VISA";
    if(/^3[4|7]/.test(N))
      return "MASTERCARD";
    else return "AMAX";
  } else return "Invalid";
}

credit(4003600000000014);

